internal/models/user.go

```diff
@@ -98,6 +99,8 @@ func (u *User) FromFormData(r *http.Request) {
 
 // Validate checks if a User has all the necessary fields.
 func (u *User) Validate() error {
+	u.Email = strings.ToLower(u.Email)
+
 	if u.Email == "" {
 		return ErrEmptyEmail
 	} else if u.Password == "" {
```

Comment:
If we want, we can set some preliminary checks such as length of password here. Not sure how necessary this is.